include/deal.II/sundials/n_vector.templates.h

```diff
@@ -279,7 +279,12 @@ namespace SUNDIALS
       inline void *
       get_communicator_as_void_ptr(N_Vector v)
       {
+#  ifndef DEAL_II_WITH_MPI
+        (void)v;
+        return nullptr;
+#  else
```

Comment:
@masterleinad fyi: this isn't compatible with the test suite for versions > 5.0.0 (which can currently only run on my local machine...) and violates a SUNDIALS contract.
From the SUNDIALS guide on `N_VGetCommunicator`:
> Returns a pointer to the MPI_Comm object associated with the vector (if applicable). For MPI-unaware vector
implementations, this should return NULL.

Problem is that the serial vector will return MPI_COMM_SELF, which is not the nullptr (if MPI is enabled). I mentioned that here https://github.com/dealii/dealii/pull/11701#discussion_r571460766 but somehow forgot when @bangerth brought up the idea with MPI_COMM_SELF :blush:

I will provide a fix.